custom_components/ha_washdata/profile_store.py

```
@@ -4501,6 +4546,83 @@ def get_envelope(self, profile_name: str) -> JSONDict | None:
+        if len(offsets) < 2 or len(env_time) < 2:
+            return list(powers)
+        diffs = np.diff(np.asarray(env_time, dtype=float))
+        pos = diffs[diffs > 0]
+        if pos.size == 0:
+            return list(powers)
+        step = float(np.median(pos))
+        span = float(env_time[-1])
+        t_max = min(float(offsets[-1]), 2.0 * span) if span > 0 else float(offsets[-1])
+        if step <= 0 or t_max <= 0:
+            return list(powers)
+        n = max(2, int(round(t_max / step)) + 1)
```

Ruff (0.16.1)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)